src/main/java/com/google/cloud/genomics/utils/ShardUtils.java

```diff
@@ -285,13 +294,47 @@ public StreamReadsRequest apply(String readGroupSetId) {
     return contigs;
   }
 
-  private static List<Contig> getAllShardsInVariantSet(String variantSetId,
+  private static List<Contig> getAllShardsInReadGroupSet(String readGroupSetId,
       SexChromosomeFilter sexChromosomeFilter, long numberOfBasesPerShard,
       GenomicsFactory.OfflineAuth auth) throws IOException, GeneralSecurityException {
-    List<Contig> contigs = getContigsInVariantSet(variantSetId, sexChromosomeFilter, auth);
+    List<Contig> contigs = getContigsInReadGroupSet(readGroupSetId, sexChromosomeFilter, auth);
     return ShardUtils.getAllShardsForContigs(contigs, numberOfBasesPerShard);
   }
-
+  
+  /**
+   * Retrieve the list of all the reference names and their start=0/end positions for the referenceSet
+   * associated with the readGroupSet.
+   * 
+   * Note that start is hardcoded to zero since references only specify their length. 
+   * 
+   * @param readGroupSetId - The id of the readGroupSet to query.
+   * @param sexChromosomeFilter - An enum value indicating how sex chromosomes should be
+   *        handled in the result.
+   * @return The list of all references in the readGroupSet.
+   * @throws IOException
+   * @throws GeneralSecurityException 
+   */
+  private static List<Contig> getContigsInReadGroupSet(String readGroupSetId,
+      SexChromosomeFilter sexChromosomeFilter, GenomicsFactory.OfflineAuth auth)
+          throws IOException, GeneralSecurityException {
+    String referenceSetId = GenomicsUtils.getReferenceSetId(readGroupSetId, auth);
+    if(null == referenceSetId) {
+      throw new IllegalArgumentException("No referenceSetId associated with readGroupSetId "
```

Comment:
Note: there is an alternative method for getting all of the references and their boundaries, since this may not be populated for all read group sets, which is to call the ListCoverageBuckets API with just a `readGroupSetId`.